### PR TITLE
Make it possible to set explicit schema i.s.o letting AutoLoader infer the schema

### PR DESCRIPTION
**Pull request by BrendBraeckmans**
## Description
Make it possible to set explicit schema i.s.o letting AutoLoader infer the schema.
Optional `schema` argument has been added

## Related Issue
[koheesio-39](https://github.com/Nike-Inc/koheesio/issues/39)

## Motivation and Context
The current implementation of AutoLoader within koheesio will always infer the schema from the files it reads. In a lot of cases this is unnecessary and might even give issues if the input data doesn't contain the required fields.

## How Has This Been Tested?
- Through UTs
- On DBX

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/koheesio/spark/readers/databricks/autoloader.py [hunk 1]
```diff
@@ -87,6 +92,10 @@ class AutoLoader(Reader):
         description="Extra inputs to provide to the autoloader. For a full list of inputs, "
         "see https://docs.databricks.com/ingestion/auto-loader/options.html",
     )
+    schema: Optional[Union[Path, str, StructType]] = Field(
+        default=None,
+        description="Explicit schema to infer the schema of the input files.",
+    )
```
Comment (dannymeijer) [inline src/koheesio/spark/readers/databricks/autoloader.py:98]: unfortunately, schema is a reserved name in Pydantic. You can use 'schema' as alias though. Perhaps do this instead:

```python
schema_: Optional[Union[Path, str, StructType]] = Field(
        default=None,
        description="Explicit schema to infer the schema of the input files.",
        alias="schema",
    )
```

with this, the outward API still uses 'schema' - internally in the class you'll have to use `self._schema` though.

File: src/koheesio/spark/readers/databricks/autoloader.py [hunk 2]
```diff
@@ -96,6 +105,22 @@ def validate_format(cls, format_specified):
...
+        schema = schema_specified
+        if isinstance(schema, StructType):
+            return schema
+
+        if schema is not None:
+            schema_path = Path(schema)
+            if schema_path.exists():
+                schema = schema_path.read_text()
+            else:
+                raise FileNotFoundError(f"Schema file not found at path {schema}")
```
Comment (dannymeijer) [inline src/koheesio/spark/readers/databricks/autoloader.py:120]: swap this if-statement around:

```python
if not schema_path.exists():
    raise FileNotFoundError(f"Schema file not found at path {schema}")
schema_path = Path(schema)
```

can also use walrus operator here:
```python
if not (schema_path := Path(schema)).exists():
    ...
```

**Review by dannymeijer — COMMENTED** [review]
See comments.

File: src/koheesio/spark/readers/databricks/autoloader.py [hunk 3]
```diff
@@ -96,6 +105,22 @@ def validate_format(cls, format_specified):
...
+            schema = StructType.fromJson(json.loads(schema))
```
Comment (maxim-mityutko) [inline src/koheesio/spark/readers/databricks/autoloader.py:121]: I think it might be benefitial to wrap this into `try...except`

Comment (BrendBraeckmans) [reply src/koheesio/spark/readers/databricks/autoloader.py:121]: Do you prefer the implementation in [rest_api](https://github.com/Nike-Inc/koheesio/blob/main/src/koheesio/spark/readers/rest_api.py#L109) to keep things in line ?

File: src/koheesio/spark/readers/databricks/autoloader.py (hunk 2 above)
Comment (dannymeijer) [reply src/koheesio/spark/readers/databricks/autoloader.py:120]: yes, let's go with that one.

File: src/koheesio/spark/readers/databricks/autoloader.py (hunk 3 above)
Comment (maxim-mityutko) [reply src/koheesio/spark/readers/databricks/autoloader.py:121]: Sure. Lets keep it consistent.

**Review by dannymeijer — APPROVED** [review]

**Review by maxim-mityutko — APPROVED** [review]
lgtm

**Pull request merged**